scripts/alert_emails/misc/generate-historical-stats.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Script to go through all the historical sent emails and generate aggregated
+ * stats about emails sent. Stats are incrementally stored to a local file (to
+ * support resuming progress in case an error is reported) and then stored to
+ * Firestore at the end.
+ */
+
+import { getFirestore } from '../../common/firebase';
+import FirebaseFirestore from '@google-cloud/firestore';
+import path from 'path';
+import * as fs from 'fs';
+
+// File to store stats to. This is also used to support resuming in case there's an error.
+const STATS_FILE = path.join(__dirname, 'stats.json');
```

Comment:
FWIW- I built in this persistence / resume logic because in the past I've had trouble reading / processing all of our email alert docs on prod in one go... But this time when I ran it on prod it ran to the end with no problems, so maybe it wasn't really necessary. 🤷